### ISSUE TITLE
renderComponent() replace (recycled?) child nodes with wrong reference

### ISSUE THREAD

**Issue by dk00** [developit/preact#337]
This [pen] reproduces the bug: click 'Load some data', then click it again.

[pen]: http://codepen.io/anon/pen/LRzGpW

The component renders
```html
<span>Still loading</span>
````
at beginning, then rerenders this(with an error): 

```html
<div title="bag">                        <!-- base -->
  <div>                                  <!-- baseParent -->
    Banana
    <span class="pull-xs-right">1</span> <!-- initialBase -->
  </div>
  <div>
    Apple
    <span class="pull-xs-right">1.3</span>
  </div>
</div>
```

The exception occurs at [`baseParent.replaceChild(base, initialBase)`][here], subsequent renderings also break.

[here]: https://github.com/developit/preact/blob/7ade27579ffce259fa34be58ef48e6bbdd1919b4/src/vdom/component.js#L157

Firefox says:
```
HierarchyRequestError: Node cannot be inserted at the specified point in the hierarchy
```

Chrome says:
```
Uncaught (in promise) DOMException: Failed to execute 'replaceChild' on 'Node': The new child element contains the parent.
```

If I wrap 'Still loading' by `p` (or anything other than `span`? to prevent reusing it), every thing works fine.

**Comment by developit** [comment]
I think I have a fix for this, will test and let you know. Thanks for the repro!

**Comment by developit** [comment]
Confirmed, this will be fixed in 6.2! 🎉

**Comment by developit** [comment]
Fix released in 6.2.
